### PR TITLE
Added pre-commit job for ovs-operator

### PR DESCRIPTION
**Pull request by amolkahat**
This patch adds pre-commit job for
ovs-operator. This job is optional.

Signed-off-by: Amol Kahat <amolkahat@gmail.com>

### CODE REVIEW

**Comment by amolkahat** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ovs-operator-main-precommit-check

**Comment by amolkahat** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ovs-operator-main-precommit-check

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ovs-operator-main-precommit-check

**Comment by Sandeepyadav93** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ovn-operator-main-precommit-check

**Comment by Sandeepyadav93** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ovs-operator-main-precommit-check

**Review by viroel — CHANGES_REQUESTED** [review]

**Comment by viroel** [comment]
Also depends-on:
https://github.com/openstack-k8s-operators/ovs-operator/pull/92
and
https://github.com/openstack-k8s-operators/ovs-operator/pull/91

**Review by marios — COMMENTED** [review]
looks like needs some updates 

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/36013/rehearse-36013-pull-ci-openstack-k8s-operators-ovs-operator-main-precommit-check/1643575331338063872

 /go/pkg/mod/sigs.k8s.io/json@v0.0.0-20220713155537-f223a00ba0e2/internal/golang/encoding/json/decode.go:1261:42: predeclared any requires go1.18 or later (-lang was set to go1.16; check go.mod)

File: ci-operator/config/openstack-k8s-operators/ovs-operator/openstack-k8s-operators-ovs-operator-main.yaml [hunk 1]
```diff
@@ -98,6 +103,18 @@ tests:
   optional: true
   steps:
     workflow: openstack-k8s-operators-test-build-deploy-kuttl
+- as: precommit-check
+  commands: |
+    export SKIP=no-commit-to-branch
+    export GOFLAGS=" -mod=readonly"
```
[the remark this answers is not in the document]
Comment (bshewale) [reply ci-operator/config/openstack-k8s-operators/ovs-operator/openstack-k8s-operators-ovs-operator-main.yaml]: Done

**Pull request merged**